src/commands/howto.ts

```diff
@@ -0,0 +1,181 @@
+- Create a thread to keep the discussion focused on your topic (react with :thread: \`:thread:\` or use \`/howto create_thread\` for more info).
+- Explain what kata you are trying to solve. Post its title, and a link to its description (react with :link: \`:link:\` or use \`/howto kata_link\` for more info).
+- Explain what your problem is. Tests do not accept your answers? Solution is timing out? Is there some other problem?
+- Post **properly formatted** code of your solution (react with :hash: \`:hash:\` or use \`/howto format_code\` for more info).
+- If the code you posted contains spoilers for the discussed kata, please delete it after discussing your matter and getting all necessary help.`,
+    reactions: [
+      { emoji: "🧵", command: "create_thread" },
+      { emoji: "🔗", command: "post_link" },
+      { emoji: "#️⃣", command: "format_code" },
+    ],
+  },
+  {
+    name: "post_link",
+    description: "How to post links to kata",
+    body: `# How to post links to kata
+- Surround the links to kata with angle brackets: \`<https://www.codewars.com/kata/50654ddff44f800200000004/python>\`. This will prevent Discord from inserting embeds into your message, as they do not carry too much useful information, and are quite annoying.
+- Please, do _NOT_  post direct links to kata trainers: make sure you don't have a trailling ~~\`/trainer\`~~ in the url.
+- Remember to also mention a title of the kata you are linking to.`,
+    reactions: [],
+  },
+  {
+    name: "create_thread",
+    description: "How to create discord threads",
+    body: `# How to create Discord threads
+Please use Discord threads to keep discussions focused, avoid spoilers, and to not introduce unnecessary noise into main channels. Remember to give a meaningful title to your thread, for example _"hobovsky's thread - help to solve Multiply kata"_, or something similar.
+
+See Discord docs if you do not know how to create threads: <https://support.discord.com/hc/en-us/articles/4403205878423-Threads-FAQ#h_01GDXVYE6AZA3QRSB42F5AGGYG>`,
+    reactions: [],
+  },
+];
+
+export const data = async () => {
+  const addTargetUserParam = (scb: SlashCommandSubcommandBuilder) =>
+    scb.addUserOption((b) =>
+      b.setName("user").setDescription("User to send the message to").setRequired(false)
+    );
+
+  let rootCommandBuilder = new SlashCommandBuilder()
+    .setName("howto")
+    .setDescription("HOWTOs and tutorials");
+
+  for (let subcommand of commands) {
+    rootCommandBuilder.addSubcommand((scb) =>
+      addTargetUserParam(scb.setName(subcommand.name).setDescription(subcommand.description))
+    );
+  }
+  return rootCommandBuilder.toJSON();
+};
+
+const allowedRoles = new Set(["admin", "mods", "mods+", "power-users"]);
+
+const hasSufficientPrivilege = (member: GuildMember | APIInteractionGuildMember | null) => {
```

Comment:
Instead of checking the user's role after the command is triggered like this, we've been configuring the command permission on the server. This way, users without the permission won't even see the command suggested.

So disable the command by default by calling `setDefaultPermission(false)` on the command builder. I'll update the permission when it's deployed.

You can find the config on your dev server under `Server Settings > Apps > Integrations > Codewars`.

There used to be an API to do this on start, but it broke (#71).

Reply:
The thing is that in my idea, permissions to run the command depend on parameters. All users are allowed to run the command and message themselves. Only privileged users are allowed to message other users with the command, to avoid spam waves from the whole noob collective when someone writes an ugly message.

Maybe this mechanism is not necessary, then I can change it. But if we want to keep this behavior, I think that permissions configured on server are not flexible enough?

Reply:
Yeah, I don't think we can set the permission based on the argument used.